src/udev/udevd.c

```diff
@@ -399,8 +399,10 @@ static int worker_lock_block_device(sd_device *dev, int *ret_fd) {
 
         fd = open(val, O_RDONLY|O_CLOEXEC|O_NOFOLLOW|O_NONBLOCK);
         if (fd < 0) {
-                log_device_debug_errno(dev, errno, "Failed to open '%s', ignoring: %m", val);
-                return 0;
+                bool ignore = IN_SET(errno, ENOENT, ENXIO, ENODEV);
```

Comment:
i wonder if we should add ERRNO_IS_DEVICE_ABSENT() as a shortcut helper for this. Sounds like something we could use more than once